src/test/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildTriggerStepRestartTest.java

```diff
@@ -64,6 +78,81 @@ public void restartBetweenJobs() throws Throwable {
         sessions.then(j -> j.jenkins.getItemByFullName("test1", FreeStyleProject.class).getBuildByNumber(1).delete());
     }
 
+    @Test
+    public void downstreamBuildActionUpstreamCompletesBeforeDownstreamStarts() throws Throwable {
+        sessions.then(j -> {
+            FreeStyleProject downstream = j.createFreeStyleProject("downstream");
+            downstream.setAssignedLabel(Label.parseExpression("agent"));
+            WorkflowJob upstream = j.jenkins.createProject(WorkflowJob.class, "upstream");
+            upstream.addProperty(new DurabilityHintJobProperty(FlowDurabilityHint.PERFORMANCE_OPTIMIZED)); // TODO: Unnecessary once we pick up https://github.com/jenkinsci/workflow-cps-plugin/pull/807.
```

Comment:
Not strictly relevant anymore, but seems best to test the lazier implementation.

Reply:
Why not just update the `bom` now to pick up https://github.com/jenkinsci/workflow-cps-plugin/pull/807?

Reply:
Because this was trivial and didn't require me to look at anything else. Looking now at dependency versions it does seem that we have the same Jenkins minimum version requirements in both plugins so I can update the BOM.